src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/service/NotificationService.kt

```diff
@@ -2,24 +2,129 @@ package uk.gov.justice.digital.hmpps.notificationsalertsvsip.service
+  @Value("\${notify.template-id.visit-cancel:}") private val visitCancelTemplateId: String,
+  @Value("\${notify.template-id.visit-cancel-no-prison-number:}") private val visitCancelNoPrisonNumberTemplateId: String,
+) {
   private companion object {
     val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+    private const val SMS_DATE_PATTERN = "dd MMMM yyyy"
+    private const val SMS_TIME_PATTERN = "hh:mm a"
+    private const val SMS_DAY_OF_WEEK_PATTERN = "EEEE"
+  }
+
+  enum class VisitEventType { BOOKED, UPDATED, CANCELLED }
+
+  fun sendMessage(visitEventType: VisitEventType, bookingReference: String) {
+    LOG.info("Visit booked event with reference - $bookingReference")
+
+    val visit = getVisit(bookingReference)
+
+    visit?.visitContact?.telephone?.let { telephoneNumber ->
+      val prisonName = getPrisonName(visit.prisonCode)
+      when (visitEventType) {
+        VisitEventType.BOOKED, VisitEventType.UPDATED -> smsSenderService.sendSms(
+          getNotificationTemplateId(visitEventType),
+          telephoneNumber,
+          mapOf(
+            "prison" to prisonName,
+            "time" to getFormattedTime(visit.startTimestamp.toLocalTime()),
+            "dayofweek" to getFormattedDayOfWeek(visit.startTimestamp.toLocalDate()),
+            "date" to getFormattedDate(visit.startTimestamp.toLocalDate()),
+            "ref number" to visit.reference,
+          ),
+        )
+
+        VisitEventType.CANCELLED -> {
+          val prisonContactNumber = getPrisonSocialVisitsContactNumber(visit.prisonCode)
+
+          if (prisonContactNumber != null) {
+            smsSenderService.sendSms(
+              visitCancelTemplateId,
+              telephoneNumber,
+              mapOf(
+                "prison" to getPrisonName(visit.prisonCode),
+                "time" to getFormattedTime(visit.startTimestamp.toLocalTime()),
+                "date" to getFormattedDate(visit.startTimestamp.toLocalDate()),
+                "ref number" to visit.reference,
+                "prison phone number" to prisonContactNumber,
+              ),
+            )
+          } else {
+            smsSenderService.sendSms(
+              visitCancelNoPrisonNumberTemplateId,
+              telephoneNumber,
+              mapOf(
+                "prison" to getPrisonName(visit.prisonCode),
+                "time" to getFormattedTime(visit.startTimestamp.toLocalTime()),
+                "date" to getFormattedDate(visit.startTimestamp.toLocalDate()),
+                "ref number" to visit.reference,
+              ),
+            )
+          }
+        }
+      }
+    }
+  }
+
+  private fun getVisit(bookingReference: String): VisitDto? {
+    return try {
+      visitSchedulerClient.getVisitByReference(bookingReference)
+    } catch (e: Exception) {
+      // if there was an error getting the visit return null
+      LOG.error("no visit found for visit with booking reference - $bookingReference")
```

Comment:
what needs to happen if the visit is not found? is it ok if an SMS is not sent in this case?

Reply:
I think a log is the only thing you can do, if it's not there,.. may be tie it down to the type of exception 404 log and forget other issues throw exception on.